src/common/hook/usePublishQueue/utils/publish.tsx

```diff
@@ -77,14 +77,25 @@ export const getRawDocuments = async (
       const contractAddress =
         formData.issuers[0]?.documentStore || formData.issuers[0]?.tokenRegistry;
       const payload = formConfig.type === "TRANSFERABLE_RECORD" ? { ownership } : {};
-      return {
-        type: formConfig.type,
-        extension,
-        contractAddress,
-        rawDocument: formData,
-        fileName,
-        payload,
-      };
+
+      if (extension) {
+        return {
+          type: formConfig.type,
+          extension,
```

Comment:
extension: extension ?? DEFAULT_ONE ?

Reply:
I realise Isaac updated it and modularised that particular line into a `generateFileName`, so the default value is actually in [this particular line](https://github.com/Open-Attestation/document-creator-website/pull/1/files#diff-e650039cbc135fbb14c393748180b577R19)

(context: I regularly fetch upstream from TradeTrust's doc creator)

Reply:
so just pass extension and if it's undefined it's fine ?

Reply:
Yes I made `extension` optional [in the types definition](https://github.com/Open-Attestation/document-creator-website/pull/1/files#diff-7d114b7dab205c5679e12568300f0221R14), so that it's handled in `PublishedTag` that defaults to `.tt`.
However, I'm not exactly sure if my design is proper?

Reply:
then why using if/else one with extension, one without

Reply:
cos there were a few test cases that were not expecting `extension` to appear

Reply:
can do 
```
return {
          type: formConfig.type,
          ...(extension? {extension} : {}),
          contractAddress,
          rawDocument: formData,
          fileName,
          payload,
        };
```

Reply:
omg........ TIL (i tried googling but probably used the wrong keywords..) thank you!!